### PR TITLE
Feat: first approach unbound action

### PR DESCRIPTION
**Pull request by FrankEssenberger**
## Context

Adds the object to create the unbound actions. Generator parts needs to be done next. Then I will also update the release notes.

## Definition of Done

Please consider all items and remove only if not applicable.

- [X] Tests created/adjusted for your changes.
~~Release notes updated.
  * Provide sufficient context so that each entry can be understood on its own.
  * Be specific about names of functions, classes, modules, etc.
  * Describe when or where this is relevant
  * Use indicative and present tense. For example, write "Provide function `name` that does X in order to Y" over "Now X can be done by calling a new function".~~
- [X] PR title adheres to [conventional commit guidelines](https://www.conventionalcommits.org) (please note that only `fix:` and `feat:` will end up in the release notes)
- [x] If applicable: Properly documented (JSDoc of public API)
- [x] If applicable: Check if `node run doc` still works.

### CODE REVIEW

File: packages/core/src/odata/v4/request/odata-action-import-request-config.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import { ODataRequestConfig } from '../../common/request/odata-request-config';
+
+export class ODataActionImportRequestConfig<
+  ParametersT
+> extends ODataRequestConfig {
+  /**
+   * Creates an instance of ODataActionImportRequestConfig.
+   *
+   * @param defaultServicePath - Default path of the service
+   * @param actionImportName - The name of the function import.
```
Comment (marikaner) [inline packages/core/src/odata/v4/request/odata-action-import-request-config.ts:15]:
```suggestion
   * @param actionImportName - The name of the action import.
```
Maybe also check other places ;)

**Comment by FrankEssenberger** [comment]
Discussed with Marika and we will use a base class for the request builders. The parameters get just a export with additional name. The methods for function and action in v4 are fixed to get and post.

File: packages/core/test/request-builder/action-import-request-builder-v4.spec.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import { defaultDestination } from '../test-util/request-mocker';
+import { ActionImportRequestBuilder } from '../../src/odata/v4/request-builder/action-import-request-builder';
+import { transformReturnValueForUndefined } from '../../src/odata/v2/request-builder';
+import { mockCsrfTokenRequest } from '../../../../test-packages/integration-tests/test/test-util/request-mocker';
+import { Destination } from '../../src/scp-cf';
+
+const servicePath = '/sap/opu/odata/sap/API_TEST_SRV';
+const host = 'https://example.com';
+const actionName = 'myActionReturnVoid';
+// TODO in the next PR the actions will be generated from the edmx and called from the generated code
```
Comment (marikaner) [inline packages/core/test/request-builder/action-import-request-builder-v4.spec.ts:12]: I hope this won't be forgotten ;)

File: packages/generator/src/function-import/function.ts [hunk 3]
```diff
@@ -51,13 +51,37 @@ function getFunctionImportStatements(
...
 
-  const returnStatement = `return new FunctionImportRequestBuilder('${
-    functionImport.httpMethod
-  }', '${service.servicePath}', '${
-    functionImport.originalName
-  }', (data) => ${responseTransformerFunctionName(
-    functionImport.returnType
-  )}(data, ${functionImport.returnType.builderFunction}), params);`;
+  const parameters =
+    service.oDataVersion === 'v2'
```
Comment (marikaner) [inline packages/generator/src/function-import/function.ts:55]: Wasn't there a function for this, like `isODataV2`?

Comment (FrankEssenberger) [reply packages/generator/src/function-import/function.ts:55]: I only found the one action on edmx metadata. I will have a look.

Comment (FrankEssenberger) [reply packages/generator/src/function-import/function.ts:55]: I only see the isV2Metadata which is  not right here.

File: packages/core/test/request-builder/action-import-request-builder-v4.spec.ts (hunk 2 above)
Comment (FrankEssenberger) [reply packages/core/test/request-builder/action-import-request-builder-v4.spec.ts:12]: sure

File: packages/core/src/odata/common/request-builder/action-and-function-import-request-builder-base.ts [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+  RequestMethodType
+} from '../../common';
+import { ActionImportParameters } from '../../v4/request/action-import-parameter';
+
+/**
+ * Create OData request to execute a action or function import.
+ * @typeparam ParametersT - Type of the function import parameters
+ * @typeparam ReturnT - Type of the function import return value
+ */
+export abstract class ActionAndFunctionImportRequestBuilderBase<
```
[the remark this answers is not in the document]
Comment (FrankEssenberger) [reply packages/core/src/odata/common/request-builder/action-and-function-import-request-builder-base.ts:22]: I remove the And

File: packages/core/src/odata/common/request-builder/action-and-function-import-request-builder-base.ts [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+  // reuse
```
[the remark this answers is not in the document]
Comment (FrankEssenberger) [reply packages/core/src/odata/common/request-builder/action-and-function-import-request-builder-base.ts]: done

**Pull request merged**